addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -158,6 +193,58 @@ weSnippetEditor.SnippetsMenu.include({
         this._super(...arguments);
         this.$('.o_we_customize_theme_btn').toggleClass('active', tab === this.tabs.THEME);
     },
+    /**
+     * Returns the animated text element wrapping the selection if it exists.
+     * 
+     * @private
+     * @param {Node} editable
+     */
+    _getAnimatedTextElement(editable) {
+        const animatedTextNode = getTraversedNodes(editable).find(n => $(n.parentElement).closest(".o_animated_text").length);
+        return $(animatedTextNode).closest('.o_animated_text');
+    },
+    /**
+     * @override
+     */
+    _addToolbar() {
+        this._super(...arguments);
+        $('#o_we_editor_toolbar_container > we-title > span').after($(`
+            <div class="btn fa fa-fw fa-2x o_we_highlight_animated_text d-none
+                ${$('body').hasClass('o_animated_text_highlighted') ? 'fa-eye text-success' : 'fa-eye-slash'}"
+                title="Highlight Animated Text"
+                aria-label="Highlight Animated Text"/>`)
+        );
+        this._toggleAnimatedTextButton();
+        this._toggleHighlightAnimatedTextButton();
+    },
+    /**
+     * Activates the button to animate text if the selection is in an
+     * animated text element or deactivates the button if not.
+     * 
+     * @private
+     */
+    _toggleAnimatedTextButton() {
+        const editable = this.options.wysiwyg.$editable[0];
+        const sel = window.getSelection();
+        if (!sel.rangeCount || !editable.contains(sel.anchorNode)) {
+            return false;
+        }
+        const $animatedText = this._getAnimatedTextElement(editable);
+        const isAnimatedText = $animatedText !== undefined && $animatedText.length > 0;
+        $('.o_we_animate_text').toggleClass('active', isAnimatedText);
+        this.$currentAnimatedText = isAnimatedText ? $animatedText : $();
+    },
+    /**
+     * Displays the button that allows to highlight the animated text if there
+     * is animated text in the page.
+     * 
+     * @private
+     */
+    _toggleHighlightAnimatedTextButton() {
+        const $animatedText = this.getEditableArea().find('.o_animated_text');
+        $('#o_we_editor_toolbar_container .o_we_highlight_animated_text').toggleClass('d-none', !$animatedText.length);
```

Comment:
Needed to search in the whole DOM? :thinking:

Reply:
To be clear, you mean "search" about o_animated_text or about  the o_we_highlight_animated_text button ?

Reply:
The line I commenting: o_we_highlight_animated_text. Also you don't search the whole DOM for o_animated_text